liquidity-rewards/test/UniswapPoolRewardsTest.js

```diff
@@ -123,13 +121,156 @@ contract('UniswapPoolRewards', function ([_, wallet1, wallet2, wallet3, wallet4,
-      console.log("calculated reward wallet1:", halfRewardMustPaid.add(earnedDiff).div(_1e18).toString());
-      console.log("calculated reward wallet2:", halfRewardMustPaid.sub(earnedDiff).div(_1e18).toString());
+      console.log("calculated reward wallet1:", halfRewardMustPaid.add(rewardsEarnedDiff).div(_1e18).toString());
+      console.log("calculated reward wallet2:", halfRewardMustPaid.sub(rewardsEarnedDiff).div(_1e18).toString());
       console.log("------------------------------------------------------------------------");
       console.log("total reward must paid:", rewardMustPaid.div(_1e18).toString());
-      console.log("earned diff:", earnedDiff.mul(new BN(2)).div(_1e18).toString());
+      console.log("rewardsEarned diff:", rewardsEarnedDiff.mul(new BN(2)).div(_1e18).toString());
       console.log("wallet2 deposit time difference from wallet1 (seconds):", timeDiff.toString());
       console.log("------------------------------------------------------------------------");
     });
+
+    it('Two depositor with the different (1:3) deposits wait for a DURATION', async function () {
+      expect(await this.pool.rewardPerToken()).to.be.bignumber.almostEqualDiv1e18('0');
+      expect(await this.pool.rewardsEarned(wallet1)).to.be.bignumber.equal('0');
+      expect(await this.pool.rewardsEarned(wallet2)).to.be.bignumber.equal('0');
+      const deposit1 = new BN(web3.utils.toWei('1'));
+      await this.pool.deposit(deposit1, { from: wallet1, gas: 1000000 });
+      const depositTime1 = await time.latest();
+
+      // Time goes by... so slowly.
+      await time.increaseTo(depositTime1.add(AMONTH.div(new BN(2)))); // Increase time by 2 weeks.
+
+      const deposit2 = new BN(web3.utils.toWei('3'));
+      await this.pool.deposit(deposit2, { from: wallet2, gas: 1000000 });
+      const depositTime2 = await time.latest();
+
+      const timeDiff = depositTime2.sub(depositTime1);
+      const testTime = depositTime2.add(AMONTH.div(new BN(2))); // Increase time by a month.
+      
+      await time.increaseTo(testTime);
+      
+      const rewardPerToken1 = this.rewardRate.mul(timeDiff).mul(_1e18).div(deposit1);
+      const rewardPerToken2 = this.rewardRate.mul(testTime.sub(depositTime2)).mul(_1e18).div(deposit1.add(deposit2));
+      const rewardPerToken = rewardPerToken1.add(rewardPerToken2);
+
+      expect(await this.pool.rewardPerToken()).to.be.bignumber.almostEqualDiv1e18(rewardPerToken);
+      expect(await this.pool.rewardsEarned(wallet1)).to.be.bignumber.almostEqualDiv1e18(rewardPerToken1.add(rewardPerToken2).mul(deposit1).div(_1e18));
+      expect(await this.pool.rewardsEarned(wallet2)).to.be.bignumber.almostEqualDiv1e18(rewardPerToken2.mul(deposit2).div(_1e18));
+      console.log("------------------------------------------------------------------------");
+      console.log("(calculated for a duration of one month)");
+      console.log("rewardRate:", this.rewardRate.toString());
```

Comment:
To make the log output from the tests a bit more human-readable, would it be possible to include the decimal points in the numbers?

I think these numbers are 18 decimal points?:

```
------------------------------------------------------------------------
(calculated for a duration of one month)
rewardRate: 124666666666666666
rewardPerToken calc: 57677
rewardPerToken actual: 57677
------------------------------------------------------------------------
rewardsEarned wallet1: 57677
rewardsEarned wallet2: 0
rewardsEarned wallet3: 151781
------------------------------------------------------------------------
      ✔ Three depositors with different (1:3:5) deposits waits for different DURATION (4486ms)
```